Schema/enums and definitions generation support.

siddarthvader opened the pull request:
Now generating the schema files, enums and definitions for custom directive.

All tests are passing.

There is a test called "Writing twice to "Writing twice doesn't duplicate definitions", for this i dont run the whole pipeline(runPipeline fn*) two times, i only run the schema definition generatio.

runPipeline Fn does not fully represent the series of operation when i run "houdini generate and since we are only testing the compiler layer i only ran schemaDefinition twice to assume it would be sufficient condition, i may be wrong.

siddarthvader committed:
schema generation working

schema generation for internal directives working

siddarthvader committed:
clean

siddarthvader committed:
added enum runtime generation and document.gql generation

tests remaining for enum runtime

siddarthvader committed:
added tests for runtime enum

siddarthvader committed:
ts enum generation wroks

siddarthvader committed:
fixed tests

siddarthvader committed:
added definitiuon directory poathg to convention.go

siddarthvader committed:
all tests passing

runing only the schema genration (and not whole pipeline)  second times
for the "writing twice test" to pass.

siddarthvader committed:
removed  complex sql query

imperatively building enums file instead of using compelx sql query

siddarthvader committed:
using private structs

siddarthvader committed:
moved functions inline

siddarthvader committed:
generate definitions and enums

AlecAivazis commented on packages/houdini-core/plugin/schema/generateDefinitions_test.go:
```diff
@@ -0,0 +1,429 @@
+			assert.Nil(t, err)
+
+			switch test.Name {
+			case "Generates runtime definitions for each enum":
```
Let's use the Extra field in the test in order to keep each test case isolated. you can save the expected values in the map and then look up the value in `test.Extra` inside of `PerformTest`

siddarthvader committed:
use text.Extra pattern for runing testsq

consolidated test runner, run different assertion based on text.Extra
values,

a also using more semantic test schema

AlecAivazis commented:
When I run the generate command in the `e2e/react` project I see a bunch of empty objects:

<img width="322" height="485" alt="image" src="https://github.com/user-attachments/assets/afa4ec59-2178-43a0-8681-3f1d9ad571da" />


The old tests validated entire snapshots of the files - let's keep that structure so we can make sure that we perfectly reproduce the previous logic

siddarthvader committed:
custom keys _remove fragments and orderingfixed

siddarthvader committed:
not sorting the enu

siddarthvader committed:
getting description out of schema for enums

generated emums.js and .ts should ahve description as the orignal
document and also storing descritpiopn in sqligte

siddarthvader committed:
added desritpion to types also

added desc to types, updated conflict statement for both tables types
and enum_Values. getting the deprecaited directive out of enum and
storing it in DB along with descritpion text, updated tests to match TS
version
databse scheam updated

AlecAivazis reviewed (approved):
Looks great! Thanks for fixing the tests 👍
